Patch to fix build with Boost 1.70 obtained from
https://github.com/PointCloudLibrary/pcl/files/2975614/patch.txt
See issue https://github.com/PointCloudLibrary/pcl/issues/2931

diff --git a/segmentation/include/pcl/segmentation/supervoxel_clustering.h b/segmentation/include/pcl/segmentation/supervoxel_clustering.h
index 0fad772..bed6c1a 100644
--- segmentation/include/pcl/segmentation/supervoxel_clustering.h
+++ segmentation/include/pcl/segmentation/supervoxel_clustering.h
@@ -524,7 +524,7 @@ namespace pcl
       };
 
       //Make boost::ptr_list can access the private class SupervoxelHelper
-      friend void boost::checked_delete<> (const typename pcl::SupervoxelClustering<PointT>::SupervoxelHelper *);
+      friend void boost::checked_delete<> (const typename pcl::SupervoxelClustering<PointT>::SupervoxelHelper *) BOOST_NOEXCEPT;
 
       typedef boost::ptr_list<SupervoxelHelper> HelperListT;
       HelperListT supervoxel_helpers_;
